api/src/main/java/com/datastrato/gravitino/rel/Column.java

```diff
@@ -40,4 +41,145 @@ public interface Column {
    * @return The default value of this column, {@link Column#DEFAULT_VALUE_NOT_SET} if not specified
    */
   Expression defaultValue();
+
+  /**
+   * Create a {@link Column} instance.
+   *
+   * @param name The name of the column.
+   * @param dataType The data type of the column.
+   * @param comment The comment of the column.
+   * @param defaultValue The default value of the column. {@link Column#DEFAULT_VALUE_NOT_SET} if
+   *     null.
+   * @return A {@link Column} instance.
+   */
+  static ColumnImpl of(String name, Type dataType, String comment, Expression defaultValue) {
+    return of(name, dataType, comment, true, false, defaultValue);
+  }
+
+  /**
+   * Create a {@link Column} instance.
+   *
+   * @param name The name of the column.
+   * @param dataType The data type of the column.
+   * @param comment The comment of the column.
+   * @return A {@link Column} instance.
+   */
+  static ColumnImpl of(String name, Type dataType, String comment) {
+    return of(name, dataType, comment, true, false, DEFAULT_VALUE_NOT_SET);
+  }
+
+  /**
+   * Create a {@link Column} instance.
+   *
+   * @param name The name of the column.
+   * @param dataType The data type of the column.
+   * @return A {@link Column} instance.
+   */
+  static ColumnImpl of(String name, Type dataType) {
+    return of(name, dataType, null, true, false, DEFAULT_VALUE_NOT_SET);
+  }
+
+  /**
+   * Create a {@link Column} instance.
+   *
+   * @param name The name of the column.
+   * @param dataType The data type of the column.
+   * @param comment The comment of the column.
+   * @param nullable True if the column may produce null values.
+   * @param autoIncrement True if the column is an auto-increment column.
+   * @param defaultValue The default value of the column. {@link Column#DEFAULT_VALUE_NOT_SET} if
+   *     null.
+   * @return A {@link Column} instance.
+   */
+  static ColumnImpl of(
+      String name,
+      Type dataType,
+      String comment,
+      boolean nullable,
+      boolean autoIncrement,
+      Expression defaultValue) {
+    return new ColumnImpl(
+        name,
+        dataType,
+        comment,
+        nullable,
+        autoIncrement,
+        defaultValue == null ? DEFAULT_VALUE_NOT_SET : defaultValue);
+  }
+
+  class ColumnImpl implements Column {
+    private String name;
+    private Type dataType;
+    private String comment;
+    private boolean nullable;
+    private boolean autoIncrement;
+    private Expression defaultValue;
+
+    private ColumnImpl(
```

Comment:
Can we also add a method to validate all the fields?

Reply:
added